api/v1alpha1/executionhook_types.go

```diff
@@ -0,0 +1,176 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ExecutionHookSpec defines the desired state of ExecutionHook
+// HookActionName is copied to ExecutionHookSpec by the controller such as
+// the Snapshot Controller.
+type ExecutionHookSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// PodSelection defines how to select pods and containers to run
+	// the executionhook. If multiple pod/containers are selected, the action will executed on them
+	// asynchronously. If execution ordering is required, caller has to implement the logic and create
+	// different hooks in order.
+	// This field is required.
+	PodSelection PodSelection `json:"podSelection" protobuf:"bytes,1,opt,name=podSelection"`
+
+	// Name of the HookAction. This is required.
+	ActionName string `json:"actionName" protobuf:"bytes,2,opt,name=actionName"`
```

Comment:
Can we make this an ObjectReference to the the HookAction?

Reply:
I think this is not an ObjectReference because we want to fix the namespace and kind here, but @xing-yang might have a better idea.

Reply:
Yes. The application snapshot controller is supposed to create this ExecutionHook object dynamically and copy HookActionName to this spec.